packages/mobile/src/components/lineup-tile/LineupTileAccessStatus.tsx

```diff
@@ -72,44 +80,59 @@ export const LineupTileAccessStatus = ({
   const staticWhite = useColor('staticWhite')
   const isUSDCPurchase =
     isUSDCEnabled && isContentUSDCPurchaseGated(streamConditions)
+  const isUnlocking = gatedTrackStatus === 'UNLOCKING'
 
   const handlePress = useCallback(() => {
-    if (isUSDCPurchase) {
+    if (isUSDCPurchase && !hasStreamAccess) {
       openPremiumContentPurchaseModal(
         { contentId, contentType },
         { source: ModalSource.TrackTile }
       )
-    } else if (contentId) {
+      return
+    }
+    if (contentId) {
       dispatch(setLockedContentId({ id: contentId }))
       dispatch(setVisibility({ drawer: 'LockedContent', visible: true }))
     }
   }, [
     isUSDCPurchase,
+    hasStreamAccess,
     contentId,
     openPremiumContentPurchaseModal,
     contentType,
     dispatch
   ])
 
+  const buttonText = isUSDCPurchase
+    ? messages.price(formatPrice(streamConditions.usdc_purchase.price))
+    : isUnlocking
+    ? messages.unlocking
+    : messages.locked
+
+  const showButtonText =
+    (isUSDCPurchase && !hasStreamAccess && !isUnlocking) || !isUSDCPurchase
+
   return (
     <TouchableOpacity onPress={handlePress}>
-      <View style={[styles.root, isUSDCPurchase ? styles.usdcPurchase : null]}>
-        {gatedTrackStatus === 'UNLOCKING' ? (
+      <View
+        style={[
+          styles.root,
+          isUSDCPurchase ? styles.usdcPurchase : null,
```

Comment:
on mobile i think we can do:
`style={[
styles.root,
[styles.usdcPurchase]: isUSDCPurchase,
[styles.unlocked]: hasStreamAccess
]}`

Reply:
I reworked this to use harmony flex props only 👍